Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2719,9 +2719,9 @@ private void handleNewRequest(ServerRequest req) {
             if((req instanceof ServerRequestRegisterClose)){
                 Log.i(TAG, "Branch is not initialized, cannot close session");
                 return;
-            }
-            else{
-                initializeSession(null);
+            } else {
+                boolean isReferrable = prefHelper_.getIsReferrable() == 1;
```

Comment:
Ok  That was a logic error. Changed to make referrable unless explicitly specified by user